### PR TITLE
Knob: non-string dropdown.

### PR DESCRIPTION
**Pull request by hysw**

### CODE REVIEW

**Comment by hysw** [comment]
This change stack on top of #342

**Comment by angela28chen** [comment]
I am in favour of this change, I think it's useful to organize the name-values for entries together rather than in separate halves. It's more readable and would probably reduce error in future for if you only adjusted one half but forgot the other.

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 1]
```diff
@@ -670,7 +666,7 @@ void GraphicsBenchmarkApp::ProcessKnobs()
     const bool enableSpheres = pEnableSpheres->GetValue();
     // If the LOD is empty, assuming we skipped the setup for all sphere resources at start
     // So need to do the intial setup for all resources here
-    const bool spheresAreSetUp               = !mSphereLODs.empty();
+    const bool spheresAreSetUp               = mSpheresAreSetUp;
```
Comment (angela28chen) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:669]: What about just removing `spheresAreSetUp` and using `mSpheresAreSetUp` in the lines below.

**Review by angela28chen — APPROVED** [review]

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp (hunk 1 above)
Comment (hysw) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:669]: The logic here need to be reworked, not just renaming, i.e. `SetupSpheresPipelines` still throw away a PipelineInterface without freeing it, I'm trying to not change much code so I trace back source PR to figure out what's the intension of existing code.

Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:669]: I see

File: include/ppx/knob.h [hunk 2]
```diff
@@ -275,7 +303,12 @@ class KnobDropdown final
     // Needed for setting from flags but use is discouraged otherwise
     void SetIndex(const std::string& newValue)
```
Comment (apazylbe) [inline include/ppx/knob.h:304]: Should this be `SetIndexByValue` and `SetIndexByName` to provide flexibility? Although if there is no use for that, we can always add that later.

**Review by apazylbe — APPROVED** [review]

File: include/ppx/knob.h (hunk 2 above)
Comment (hysw) [reply include/ppx/knob.h]: Nothing uses it, so deleted instead, we can add `SetIndexBy{Name,Value}` when we needs them.

**Review by elviscapiaq — APPROVED** [review]
I like this change which would help to organize pairs of name-values. Also, it can reduce errors in the future.

**Review by wangra-google — APPROVED** [review]

**Pull request merged**